### PR TITLE
fix(multiple database): pre sql check node failed

### PR DESCRIPTION
**Pull request by zijiacj**
#### What type of PR is this?
type-bug

#### What this PR does / why we need it:
The reason for the error is that the value of the password attribute in the data source connection is null. This pr gets the data source connection with the password value for each database

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowInstanceService.java [hunk 1]
```diff
@@ -358,20 +357,23 @@ private List<FlowInstanceDetailResp> innerCreate(@NotNull @Valid CreateFlowInsta
...
             cloudMetadataClient.checkPermission(OBTenant.of(conn.getClusterName(),
                     conn.getTenantName()), conn.getInstanceType(), false, CloudPermissionAction.READONLY);
+            conns.add(conn);
         } else if (createReq.getTaskType() == TaskType.MULTIPLE_ASYNC) {
             MultipleDatabaseChangeParameters taskParameters =
                     (MultipleDatabaseChangeParameters) createReq.getParameters();
-            List<DatabaseChangeConnection> conns = taskParameters.getDatabases().stream().map(
-                    DatabaseChangeDatabase::getDataSource).distinct().collect(Collectors.toList());
+            // Gets the data source connection collection that contains the password
+            conns = taskParameters.getDatabases().stream().map(x -> x.getDataSource().getId()).distinct().map(
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowInstanceService.java:370]: O(n) complex, we may change another way to do this~

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/PreCheckRuntimeFlowableTask.java [hunk 2]
```diff
@@ -134,10 +135,12 @@ protected Void start(Long taskId, TaskService taskService, DelegateExecution exe
             throw new ServiceTaskError(new RuntimeException("Can not find task entity by id " + preCheckTaskId));
         }
         this.creatorId = FlowTaskUtil.getTaskCreator(execution).getId();
-        try {
-            this.connectionConfig = FlowTaskUtil.getConnectionConfig(execution);
-        } catch (VerifyException e) {
-            log.info(e.getMessage());
+        if (taskEntity.getTaskType() != TaskType.MULTIPLE_ASYNC) {
```
[the remark this answers is not in the document]
Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/PreCheckRuntimeFlowableTask.java]: deleted it

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/ConnectionService.java [hunk 3]
```diff
@@ -874,10 +894,28 @@ public ConnectionConfig internalGetSkipUserCheck(Long id, boolean withEnvironmen
         return config;
     }
 
+    @SkipAuthorize("odc internal usage")
+    public List<ConnectionConfig> internalGetSkipUserCheck(Collection<Long> ids, boolean withEnvironment,
+            boolean withProject) {
+        List<ConnectionEntity> entities = getEntities(ids);
```
[the remark this answers is not in the document]
Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/connection/ConnectionService.java:879]: have done

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/util/FlowTaskUtil.java [hunk 4]
```diff
@@ -256,10 +257,26 @@ public static String getSchemaName(@NonNull DelegateExecution execution) {
         return internalGet(value, String.class).orElseThrow(() -> new VerifyException("Schema name is absent"));
     }
 
+    // use setConnectionConfigList instead
+    @Deprecated
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/flow/util/FlowTaskUtil.java:261]: why deprecate it ?

there exists tasks already use connectionConfig,
connectionConfigList should only referenced by multiple database change.

Comment (yhilmare) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/flow/util/FlowTaskUtil.java:261]: `setConnectionConfigList` can cover all situations of `setConnectionConfig` and we prefer using the new one~

Comment (yizhouxw) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/flow/util/FlowTaskUtil.java:261]: list is not clear for all tasks except mulitple database change.
the commit accepted, anyway i resource my opnion.

**Review by yizhouxw — DISMISSED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**